### PR TITLE
Encryption

### PR DESCRIPTION
**Pull request by thommierother**
added changes from review

### CODE REVIEW

**Comment by thommierother** [comment]
Hi @settermjd 
I am still learning/playing how to handle atom and its packages and the branching. Sorry if this causes some confusion. There should be two PRs ( 1. from the review and, 2. additional note concerning re-sync after finishing the migration).

**Comment by settermjd** [comment]
@thommierother, not a worry. Thanks for continuing to contribute.

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc [hunk 1]
```diff
@@ -1,40 +1,46 @@
...
+NOTE: User key-based encryption is planned to be removed from ownCloud in the near future.
+While it is a bit more secure than a central encryption approach, User key-based encryption has some disadvantages.
+It blocks some additional functions such as the integration of an online editor like LibreOffice or OnlyOffice into ownCloud and can cause problems when sharing files with groups.
+Therefore Master-key-based encryption is now the recommended setup for all new installations.
 
-If you need to migrate from User Key-based to Master Key-based encryption, there are several steps that you need to follow to ensure a smooth and complete transition.
-These are:
+The decryption workflow described here works for both MySQL and SQLite based databases. It will only work when users have enabled the password recovery. Moreover, an admin recovery password needs to be activated and available for the ownCloud administrator.
```
Comment (settermjd) [inline modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: The second sentence should start with "However, ", given the preceding sentence.

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc [hunk 2]
```diff
@@ -1,40 +1,46 @@
...
+
+After re-enabling the encryption, it is possible that some or all users see a re-synchronisation of their data from the server to the desktop client, especially for shared folders. An information to the users (e.g. through the announcement app) prior to the migration process is recommended.
```
Comment (settermjd) [inline modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: This sentence is introduced too abruptly after telling the user why Master-key based encryption is the new recommended default, in place of User-key based encryption. Please either provide more information about what the user needs to do to migrate from User-key to Master-key based encryption, or link to the existing documentation which shows how to do that. 

Secondly, I'd rework the paragraph as follows:

> it is possible that some or all users to see a re-synchronisation of their data from the server to the desktop client — especially for shared folders. A notification to the users (e.g., through the announcement app) prior to the migration process is recommended. 

Please link to the Announcement App in that sentence too.

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc [hunk 3]
```diff
@@ -1,40 +1,46 @@
...
 ----
-sudo -u www-data occ encryption:decrypt-all --continue=yes && \
-  sudo -u www-data occ encryption:disable --no-interaction && \
-  sudo -u www-data occ app:disable --no-interaction encryption
+sudo -u www-data php occ encryption:decrypt-all --continue=yes && \
+  sudo -u www-data php occ encryption:disable --no-interaction && \
+  sudo -u www-data php occ app:disable --no-interaction encryption
 ----
 
+NOTE: The decryption of the files by the central administration requires the current passwords of all users!
```
Comment (settermjd) [inline modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: "central administration" should be "system administrator".

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc [hunk 4]
```diff
@@ -51,6 +57,12 @@ sqlite> delete from oc_appconfig where appid='encryption';
 sqlite> select * from oc_appconfig where appid='encryption';
 ----
 
+Same for mysql:
```
Comment (settermjd) [inline modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: mysql should be written as MySQL.

File: package.json [hunk 5]
```diff
@@ -16,7 +16,7 @@
   ],
   "dependencies": {
     "@antora/cli": "^1.1.1",
-    "@antora/site-generator-default": "^1.1.1"
+    "@antora/site-generator-default": "^2.0.0"
```
Comment (settermjd) [inline package.json]: Please revert this change.

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc [hunk 6]
```diff
@@ -51,6 +57,12 @@ sqlite> delete from oc_appconfig where appid='encryption';
...
+
+----
+SELECT * FROM `oc_appconfig` WHERE `appid` LIKE 'encryption'
```
Comment (settermjd) [inline modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: This SQL statement needs a trailing semicolon.

**Comment by thommierother** [comment]
ok, I hope taht I did not miss something ....

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc [hunk 7]
```diff
@@ -1,40 +1,50 @@
...
+NOTE: User key-based encryption is planned to be removed from ownCloud in the near future.
+While it is a bit more secure than a central encryption approach, User key-based encryption has some disadvantages.
+It blocks some additional functions such as the integration of an online editor like LibreOffice or OnlyOffice into ownCloud and can cause problems when sharing files with groups.
+Therefore Master-key-based encryption is now the recommended setup for all new installations.
 
-If you need to migrate from User Key-based to Master Key-based encryption, there are several steps that you need to follow to ensure a smooth and complete transition.
-These are:
+The decryption workflow described here works for both MySQL and SQLite based databases. However, it will only work with the following pre-conditions:
+
+ - the admin recovery password needs to be activated during creation of new accounts for all users (settings, personal, encryption).
```
Comment (settermjd) [inline modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: Sorry to be nitpicky, but please capitalise the first letter in these sentences, and change the list format marker from a hyphen to an asterisk?

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc [hunk 8]
```diff
@@ -1,40 +1,50 @@
...
+  - the admin recovery password must be  available for the ownCloud administrator
+
+It is possible that some or all users see a re-synchronisation of their data from the server to the desktop client - especially for shared folders.
```
Comment (settermjd) [inline modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: s/all users see a/all users to see a/g

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc [hunk 9]
```diff
@@ -1,40 +1,50 @@
...
 ----
-sudo -u www-data occ encryption:decrypt-all --continue=yes && \
-  sudo -u www-data occ encryption:disable --no-interaction && \
-  sudo -u www-data occ app:disable --no-interaction encryption
+sudo -u www-data php occ encryption:decrypt-all --continue=yes && \
+  sudo -u www-data php occ encryption:disable --no-interaction && \
+  sudo -u www-data php occ app:disable --no-interaction encryption
 ----
 
+NOTE: The decryption of the files by the system administration requires the current passwords of all users!
```
Comment (settermjd) [inline modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: s/the system administration/the system administrator/g

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc [hunk 10]
```diff
@@ -1,40 +1,50 @@
...
 These commands are: xref:configuration/server/occ_command.adoc#encryption[`occ encryption:decrypt-all` and `occ encryption:disable`], and xref::configuration/server/occ_command.adoc#apps-commands[`occ app:disable`].
 
 You can see an example of calling the commands listed below, configured to require no user interaction.
 
 [source,console]
 ----
-sudo -u www-data occ encryption:decrypt-all --continue=yes && \
-  sudo -u www-data occ encryption:disable --no-interaction && \
-  sudo -u www-data occ app:disable --no-interaction encryption
+sudo -u www-data php occ encryption:decrypt-all --continue=yes && \
```
Comment (mmattel) [inline modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: {occ-command-example-prefix}
https://github.com/owncloud/docs/blob/master/docs/best-practices.md#occ-examples

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc [hunk 11]
```diff
@@ -77,10 +93,10 @@ The following example shows how to do this.
 
 [source,console]
 ----
-sudo -u www-data occ app:enable encryption && \
-  sudo -u www-data occ encryption:enable && \
-  sudo -u www-data occ encryption:select-encryption-type masterkey -y && \
-  sudo -u www-data occ encryption:encrypt-all && \
+sudo -u www-data php occ app:enable encryption && \
+  sudo -u www-data php occ encryption:enable && \
```
Comment (mmattel) [inline modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: {occ-command-example-prefix}
https://github.com/owncloud/docs/blob/master/docs/best-practices.md#occ-examples

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc (hunk 10 above)
Comment (settermjd) [reply modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: Thanks. I forgot to think about this.

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc (hunk 8 above)
Comment (thommierother) [reply modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: @settermjd Sorry, I don't understand 

"s/all users see a/all users to see a/g"

What do I have to change?

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc (hunk 9 above)
Comment (thommierother) [reply modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: same question as above

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc (hunk 8 above)
Comment (settermjd) [reply modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: Worry, what that meant was to change "all users see a" to "all users to see a".

File: modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc (hunk 9 above)
Comment (settermjd) [reply modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc]: Please change "the system administration" to "the system administrator".

**Comment by settermjd** [comment]
@thommierother, can you resolve the conflicts, please?

**Comment by thommierother** [comment]
@settermjd yep, will do

**Comment by mmattel** [comment]
Closing in favor of #2199
